x-pack/plugins/observability_solution/observability_onboarding/public/application/packages_list/index.tsx

```diff
@@ -99,31 +100,30 @@ const PackageListGridWrapper = ({
                 },
               }}
               onChange={(arg) => {
-                if (setSearchQuery) {
-                  setSearchQuery(arg.queryText);
-                }
-                setIsInitialHidden(false);
+                setSearchQuery?.(arg.queryText);
+                if (isInitialHidden) setIsInitialHidden(false);
               }}
-              query={searchQuery}
+              query={searchQuery ?? ''}
             />
           </div>
         )}
-        {showPackageList && (
-          <PackageList
-            list={list}
-            searchTerm={searchQuery ?? ''}
-            showControls={false}
-            showSearchTools={false}
-            // we either don't need these properties (yet) or handle them upstream, but
-            // they are marked as required in the original API.
-            selectedCategory={selectedCategory}
-            setSearchTerm={() => {}}
-            setCategory={() => {}}
-            categories={[]}
-            setUrlandReplaceHistory={() => {}}
-            setUrlandPushHistory={() => {}}
-          />
-        )}
+        {showPackageList &&
+          ((showSearchBar && !!searchQuery) || !showSearchBar) && (
```

Comment:
I'm a bit lost in all the conditions 😅 but the second line might be redundant, `showPackageList` seems to cover it.